### PR TITLE
Fix controller not rendering for copied gamecontroller controls

### PR DESCRIPTION
**Pull request by garbear**
## Description

This PR fixes an extreme edge case of a `gamecontroller` control used in a list-within-a-list with static listitems provided by a `<content>` node. Sporadically, when updating the outer list, the controller images within the inner list can disappear, and can also sporadically return.

screenshot00006

The reason is because the control path for copying lists-within-lists does not properly clear `m_currentController`, causing the static item state to not set the filename because the new controller has the same ID as `m_currentController`.

However, CGUIImage (which `gamecontroller` derives from) properly resets image filename state when copied (as you can see in the correct "DefaultAddonNone.png" images being shown on each line). So skip the check for the existing controller ID and let CGUIImage handle filename state updates.

## Motivation and context

Helping @HitcherUK develop XML for the new `gameagents` window. He didn't have multiple controllers to test the UI with, so I showed him how to modify the window with static listitems so get skin dimensions right.

However, when scrolling the static items, I noticed all the controller images disappeared and returned sporadically. Debugging led me to this fix.

## How has this been tested?

I added static listitems using this XML:

```patch
--- a/addons/skin.estuary/xml/Includes_Games.xml
+++ b/addons/skin.estuary/xml/Includes_Games.xml
@@ -418,6 +418,44 @@
 							</control>
 							<include>GameDialogAgentLayout</include>
 						</focusedlayout>
+						<content>
+							<item>
+								<label>Player 1</label>
+							</item>
+							<item>
+								<label>Player 2</label>
+							</item>
+							<item>
+								<label>Player 3</label>
+							</item>
+							<item>
+								<label>Player 4</label>
+							</item>
+							<item>
+								<label>Player 5</label>
+							</item>
+							<item>
+								<label>Player 6</label>
+							</item>
+							<item>
+								<label>Player 7</label>
+							</item>
+							<item>
+								<label>Player 8</label>
+							</item>
+							<item>
+								<label>Player 9</label>
+							</item>
+							<item>
+								<label>Player 10</label>
+							</item>
+							<item>
+								<label>Player 11</label>
+							</item>
+							<item>
+								<label>Player 12</label>
+							</item>
+						</content>
 					</control>
 				</control>
 				<control type="scrollbar" id="61">
@@ -465,16 +503,61 @@
 			<enable>false</enable>
 			<itemlayout width="96" height="96">
 				<control type="gamecontroller">
+					<texture>$INFO[ListItem.Icon]</texture>
+					<controllerid>$INFO[ListItem.Property(controllerid)]</controllerid>
 					<peripherallocation>$INFO[ListItem.FilenameAndPath]</peripherallocation>
 					<controllerdiffuse>button_focus</controllerdiffuse>
 				</control>
 			</itemlayout>
 			<focusedlayout width="96" height="96">
 				<control type="gamecontroller">
+					<texture>$INFO[ListItem.Icon]</texture>
+					<controllerid>$INFO[ListItem.Property(controllerid)]</controllerid>
 					<peripherallocation>$INFO[ListItem.FilenameAndPath]</peripherallocation>
 					<controllerdiffuse>button_focus</controllerdiffuse>
 				</control>
 			</focusedlayout>
+			<content>
+				<item>
+					<icon>DefaultAddonNone.png</icon>
+				</item>
+				<item>
+					<property name="controllerid">game.controller.default</property>
+				</item>
+				<item>
+					<property name="controllerid">game.controller.default</property>
+				</item>
+				<item>
+					<property name="controllerid">game.controller.default</property>
+				</item>
+				<item>
+					<property name="controllerid">game.controller.default</property>
+				</item>
+				<item>
+					<property name="controllerid">game.controller.default</property>
+				</item>
+				<item>
+					<property name="controllerid">game.controller.default</property>
+				</item>
+				<item>
+					<property name="controllerid">game.controller.default</property>
+				</item>
+				<item>
+					<property name="controllerid">game.controller.default</property>
+				</item>
+				<item>
+					<property name="controllerid">game.controller.default</property>
+				</item>
+				<item>
+					<property name="controllerid">game.controller.default</property>
+				</item>
+				<item>
+					<property name="controllerid">game.controller.default</property>
+				</item>
+				<item>
+					<property name="controllerid">game.controller.default</property>
+				</item>
+			</content>
 		</control>
 	</include>
 </includes>
```

Screenshot above shows the bug.

After, showing all controllers rendering correctly:

screenshot00008

## What is the effect on users?

* Basically none, edge case is so extreme it's only seen during heavy skin development

## Types of change

- [x] **Bug fix** (non-breaking change which fixes an issue)
- [ ] **Clean up** (non-breaking change which removes non-working, unmaintained functionality)
- [ ] **Improvement** (non-breaking change which improves existing functionality)
- [ ] **New feature** (non-breaking change which adds functionality)
- [ ] **Breaking change** (fix or feature that will cause existing functionality to change)
- [ ] **Cosmetic change** (non-breaking change that doesn't touch code)
- [ ] **Student submission** (PR was done for educational purposes and will be treated as such)
- [ ] **None of the above** (please explain below)

### CODE REVIEW

**Comment by scott967** [comment]
I am attempting to skin this.  What I see on my Windows system does not match the screenshot above.  Instead in list id="5" I get a list of attached controllers.  Clicking on a controller opens the controller settings dialog.
screenshot00003

here is a log https://paste.kodi.tv/rokaceneba.kodi
